NDTensors/ext/NDTensorsMetalExt/adapt.jl

```diff
@@ -6,6 +6,6 @@ end
 
 # More general than the version in Metal.jl
 function Adapt.adapt_storage(arraytype::Type{<:MtlArray}, xs::AbstractArray)
-  arraytype_specified = specify_parameters(arraytype, get_parameters(xs))
+  arraytype_specified = specify_parameters(arraytype, parameters(xs))
```

Comment:
This is an interesting use case of `specify_parameters`. An issue I see with this current version is that the positions of the parameters may not line up, i.e. it assumes the first parameter of `xs` is the `eltype` but that may not be the case.

One way this could be written is:
```julia
specify_parameters(arraytype, (eltype, ndims, buffertype), parameters(xs, (eltype, ndims, buffertype)))
```

Reply:
It may be challenging to make that type stable though.

Reply:
This is now implemented so I am resolving this conversation.